platform/linuxbsd/wayland/display_server_wayland.cpp

```diff
@@ -1042,6 +1081,138 @@ void DisplayServerWayland::window_start_resize(WindowResizeEdge p_edge, WindowID
 	wayland_thread.window_start_resize(p_edge, p_window);
 }
 
+void DisplayServerWayland::window_set_hdr_output_enabled(const bool p_enabled, WindowID p_window) {
+#if defined(RD_ENABLED)
+	if (rendering_context) {
+		rendering_context->window_set_hdr_output_enabled(p_window, p_enabled);
+	}
+#endif
+}
+
+bool DisplayServerWayland::window_is_hdr_output_enabled(WindowID p_window) const {
+#if defined(RD_ENABLED)
+	if (rendering_context) {
+		return rendering_context->window_get_hdr_output_enabled(p_window);
+	}
+#endif
+	return false;
+}
+
+void DisplayServerWayland::window_set_hdr_output_prefer_high_precision(const bool p_enabled, WindowID p_window) {
+#if defined(RD_ENABLED)
+	if (rendering_context) {
+		return rendering_context->window_set_hdr_output_prefer_high_precision(p_window, p_enabled);
+	}
+#endif
+}
+
+bool DisplayServerWayland::window_is_hdr_output_preferring_high_precision(WindowID p_window) const {
+#if defined(RD_ENABLED)
+	if (rendering_context) {
+		return rendering_context->window_get_hdr_output_prefer_high_precision(p_window);
+	}
+#endif
+	return false;
+}
+
+void DisplayServerWayland::window_set_hdr_output_use_screen_luminance(const bool p_enabled, WindowID p_window) {
+	MutexLock mutex_lock(wayland_thread.mutex);
+
+	WaylandThread::WindowState *window = wayland_thread.window_get_window_state(p_window);
+	window->use_screen_profile = p_enabled;
+
+	//TODO: are max_cll and max_fall relevant here?
+#if defined(RD_ENABLED)
+	if (rendering_context) {
+		if (p_enabled) {
```

Comment:
You may want to consider creating your own version of `DisplayServerWindows::_update_hdr_output_for_window` that sets the HDR output state and luminance when:
1. `use_screen_luminance` changes
2. `hdr_output_enabled` changes
3. The window moves to another monitor
4. The HDR state of the display changes